plugins/connections/ui/src/main/java/org/pentaho/di/vfs/connections/ui/dialog/ConnectionDialog.java

```diff
@@ -299,6 +313,51 @@ private void updateConnectionType( String connectionType ) {
     wConnectionTypeComp.layout();
   }
 
+  private void addRootPathComponent() {
+    VFSConnectionDetails vfsConnectionDetails = asVFSConnectionDetails( connectionDetails );
+    if ( vfsConnectionDetails != null && vfsConnectionDetails.isSupportsRootPath() ) {
+      Control control = getLastWidgetFromParentComposite( wDetailsWrapperComp );
+      Label wlRootPath =  helper.createLabel( wDetailsWrapperComp, SWT.LEFT | SWT.WRAP, "ConnectionDialog.RootFolderPath.Label", control );
+      wRootPath = helper.createTextVar( variableSpace, wDetailsWrapperComp, TEXT_VAR_FLAGS, wlRootPath, 0 );
+      wRootPath.setText( Const.NVL( vfsConnectionDetails.getRootPath(), "" ) );
+      wRootPath.addModifyListener( modifyEvent -> vfsConnectionDetails.setRootPath( wRootPath.getText() ) );
+
+      Object adapter = vfsDetailsComposite.getRootPathSelectionAdapter( wRootPath );
+      if ( adapter instanceof SelectionAdapterFileDialogTextVar ) {
+        addBrowseButtonForRootPathComponent( adapter );
```

Comment:
Cast would better be done here, and not in `addBrowseButtonForRootPathComponent`.

Reply:
Addressed